File: MyApplication/app/src/main/java/com/example/gaofei/myapplication/MainActivity.java
package com.example.gaofei.myapplication;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.example.gaofei.myapplication.utils.LogUtils;
import com.example.mylibrary.Test2;

public class MainActivity extends BaseAct {
    private static final String TAG = "MainActivity";
    private ImageView mButton;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initData();
    }

    private void initData() {

    }

    private void application(){
        LogUtils.d("1");
    }

    private void initView() {
    }

    private void initView2() {
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: MyApplication/android4/src/main/java/com/example/android4/MainActivity.kt
package com.example.android4

import android.app.Activity
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.Binder
import android.os.Bundle
import android.os.Process
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : Activity() {
    companion object {
         var count = 1;
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT
        count++;
        Log.d("aaaaa", "${count} onCreate")
        button.setOnClickListener {
            val pid = Binder.getCallingPid()
            val uid = Binder.getCallingUid()
            val userHandle = Binder.getCallingUserHandle()
            val uid1 = Process.myUid()
            button.text = "${uid1} ${pid} ${uid}"

            val intent = Intent(this, MainActivity::class.java)
            startActivity(intent)
        }
    }

    override fun onPause() {
        super.onPause()
        Log.d("aaaaa", "${count} onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d("aaaaa", "${count} onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("aaaaa", "${count} onDestroy")
    }
}
